thekuuzzz/sqlite/QuarantineParse.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from urlparse import urlsplit
+
+import datetime
+import os
+import sqlite3
+
+
+class QuarantineParse(object):
+    """Parses the QuarantineEventsV2 file and prints domains + download agents
+
+        Args:
+        none
+
+        Returns:
+        none
+    """
+
+    def parseFile():
+        agent_dict = dict(agent=' ', count=0)
+        url_dict = dict(site=' ', count=0)
+        try:
+            database = sqlite3.connect('QuarantineEventsV2')
+            database_query = database.execute('SELECT LSQuarantineTimestamp AS Time,
+            LSQuarantineAgentName AS Agent, LSQuarantineOriginURLString AS URL,
+            LSQuarantineDataURLString AS Data
+            FROM LSQuarantineEvent ORDER BY Time;')
+            for record in database_query.split('|'):
+                if record.startswith(str.isdigit()):
+                    time = _convert_time(record)
+                elif record.startswith('https') | record.startswith('http'):
+                    url = record
+                    uri = urlsplit(record, scheme='', allow_fragments=True)
+                    domain = '{uri.scheme}://{uri.netloc}/'
+                    if domain not in url_dict:
+                        url_dict[domain] = 1
+                    else:
+                        url_dict[domian] = url_dict[domain]+1
+                    else:
+                        if record not in agent_dict:
+                            agent_dict[record] = 1
+                        else:
+                            agent_dict[record] = agent_dict[record]+1
+                            source = record
+            print(time + ' ' + source + ' downloaded from ' + url + '\n')
+            print('List of Agents: ' + '\n')
+            for agents, values in agent_dict:
+                print(agent + ' -> ' + values)
+            print('\n' + 'List of Domains: ' + '\n')
+            for domains, values in url_dict:
+                print(domains + ' -> ' + values)
+        except EnvironmentError as e:
```

Comment:
where are all these errors coming from? where are they raised... keep them close by...

what about sqlite3.DatabaseError ? I don't see that caught here...